### ISSUE TITLE
Blue screen when quitting Terminal during startup

### ISSUE THREAD

**Issue by amarchino** [microsoft/terminal#3268]
# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Windows Terminal version (if applicable):
Microsoft Windows [Version 10.0.18362.418]

Any other software?
WSL (version 1, not tried on WSL2 since not on Insider Program).
Terminal (version: 0.5.2762.0)
```

# Steps to reproduce

Set up WSL as the default terminal to be used in the application.
Start up the Terminal, and before the startup is complete (meaning, before the terminal is available to be interacted with), close it down.

# Expected behavior

Closing the instance of Terminal

# Actual behavior

Blue screen on the pc, with a "KMODE EXCEPTION NOT HANDLED".

I stand at disposal for whatever data might be needed.

**Comment by amarchino** [comment]
Link to the feedback: https://aka.ms/AA6dgr8
I think I actually cannot attach  come data to the feedback due to the Blue Screen (which would kill the recording)
If need be, I'll try and recover/send a dump via the proposed method.

**Comment by dclong** [comment]
Yesterday I encountered 2 blue screens with my Windows Insider Build. I believe it was caused by using WSL with Windows Terminal because without using Windows Terminal and keep all other apps running the blue screen never happened again and I did use Windows Terminal before (without WSL) and it didn't cause blue screen either. The stop code were SYSTEM_SERVICE_EXCEPTION and ATTEMPTED_WRITE_TO_READONLY_MEMORY.
